### PR TITLE
Fully incremental fetch

### PR DESCRIPTION
**Pull request by as2388**
Review #39 first.

For example usage, see https://github.com/AfricasVoices/Project-ADSS/pull/28.

### CODE REVIEW

File: rapid_pro_tools/rapid_pro_client.py [hunk 1]
```diff
@@ -22,12 +37,44 @@ def get_flow_id(self, flow_name):
...
+        Gets the raw runs for the given flow_id from RapidPro.
+        
+        :param flow_id: Id of the flow to download the runs of.
+        :type flow_id: str
+        :param range_start_inclusive: Start of the date-range to download runs from. If set, only downloads runs
+                                      last modified since that date, otherwise downloads from the beginning of time.
+        :type range_start_inclusive: datetime.pyi | None
+        :param range_end_exclusive: End of the date-range to download runs from. If set, only downloads runs
+                                    last modified before that date, otherwise downloads until the end of time.
+        :param raw_export_log: File to write the raw data downloaded during this function call to.
```
Comment (lukechurch) [inline rapid_pro_tools/rapid_pro_client.py]: consider: raw_export_log_file

And in the description add a note that it's going to be jsony.

File: rapid_pro_tools/rapid_pro_client.py [hunk 2]
```diff
@@ -37,28 +84,174 @@ def get_raw_runs_for_flow_id(self, flow_id, range_start_inclusive=None, range_en
...
         return raw_runs
 
-    def get_raw_contacts(self):
-        print("Fetching all raw contacts...")
-        raw_contacts = self.rapid_pro.get_contacts().all(retry_on_rate_exceed=True)
+    def get_raw_contacts(self, range_start_inclusive=None, range_end_exclusive=None, raw_export_log=None):
+        """
+        Gets the raw contacts from RapidPro.
+        
+        :param range_start_inclusive: Start of the date-range to download contacts from. If set, only downloads contacts
```
Comment (lukechurch) [inline rapid_pro_tools/rapid_pro_client.py]: Right - so this is modified. 

I think we might need to put this in the variable names, there are few too many different interpretations.

Comment (lukechurch) [reply rapid_pro_tools/rapid_pro_client.py]: Considered contacts_modified_range_inclusive

It's getting a bit long winded, but important I think

**Comment by as2388** [comment]
@lukechurch ready for review

File: rapid_pro_tools/rapid_pro_client.py (hunk 1 above)
Comment (as2388) [reply rapid_pro_tools/rapid_pro_client.py]: changed

File: rapid_pro_tools/rapid_pro_client.py (hunk 2 above)
Comment (as2388) [reply rapid_pro_tools/rapid_pro_client.py]: renamed to last_modified_after_inclusive etc.

**Pull request merged**